lib/TwoFactorAuth.php

```diff
@@ -111,37 +64,24 @@ public function createSecret($bits = 80, $requirecryptosecure = true)
 
     /**
      * Calculate the code with given secret and point in time
-     *
-     * @param string $secret
-     * @param ?int $time
-     *
-     * @return string
      */
-    public function getCode($secret, $time = null)
+    public function getCode(string $secret, ?int $time = null): string
     {
         $secretkey = $this->base32Decode($secret);
 
         $timestamp = "\0\0\0\0" . pack('N*', $this->getTimeSlice($this->getTime($time)));  // Pack time into binary string
-        $hashhmac = hash_hmac($this->algorithm, $timestamp, $secretkey, true);             // Hash it with users secret key
+        $hashhmac = hash_hmac($this->algorithm->value, $timestamp, $secretkey, true);             // Hash it with users secret key
         $hashpart = substr($hashhmac, ord(substr($hashhmac, -1)) & 0x0F, 4);               // Use last nibble of result as index/offset and grab 4 bytes of the result
         $value = unpack('N', $hashpart);                                                   // Unpack binary value
         $value = $value[1] & 0x7FFFFFFF;                                                   // Drop MSB, keep only 31 bits
 
-        return str_pad((string) ($value % pow(10, $this->digits)), $this->digits, '0', STR_PAD_LEFT);
+        return str_pad((string) ($value % 10** $this->digits), $this->digits, '0', STR_PAD_LEFT);
```

Comment:
I would have left this as `pow()` so it was more obvious to the casual individual who doesn't know all the shorthands

Reply:
This was made by phpcs-fixer's [pow_to_exponation](https://cs.symfony.com/doc/rules/alias/pow_to_exponentiation.html) rule, which is also used by [Symfony](https://cs.symfony.com/doc/rules/index.html). 

I don't have a strong opinion on this, but I'm pretty sure not calling the `pow()` function is faster. A casual individual won't change this part of the code anyway :p